### PR TITLE
🐛 Changed dbt packages in requirements.txt

### PR DESCRIPTION
**Pull request by adrian-wojcik**
## Summary

The PR containing changes in requirements.txt file. Dbt-sqlserver were upgraded to djagoda's fork in order to fix bug in dbt.

## Importance

It is important because without it some of the Transform&Catalog flows will be failing on Prefect due to bug in current version of dbt-sqlserver. Linked issue: #854 

## Checklist

This PR:
- [X ] follows the guidelines laid out in `CONTRIBUTING.md`
- [X ] links relevant issue(s)
- [] adds/updates tests (if appropriate)
- [] adds/updates docstrings (if appropriate)
- [X ] adds an entry in `CHANGELOG.md`

### CODE REVIEW

**Review by Rafalz13 — COMMENTED** [review]
Added comments

File: requirements.txt [hunk 1]
```diff
@@ -39,8 +39,7 @@ aiolimiter==1.0.0
 protobuf>=3.19.0, <3.20
 avro-python3==1.10.2
 pygit2>=1.10.1, <1.11.0
-dbt-core==1.3.2
-dbt-sqlserver==1.3.1
+dbt-sqlserver @ git+https://github.com/djagoda881/dbt-sqlserver.git@v1.3.latest_option_clause
```
[the remark this answers is not in the document]
Comment (adrian-wojcik) [reply requirements.txt:42]: After speaking with Dominik, it seems like we can not provide static version number as this is version on his own fork where he fixed bug with max recursion

**Review by Rafalz13 — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,9 +8,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Added
 
 ### Fixed
+- Changed requirements.txt to level up version of dbt-sqlserver in order to fix bug with `MAXRECURSION` error in dbt_run
```
Comment (Rafalz13) [inline CHANGELOG.md:11]:
```suggestion
- Changed `requirements.tx`t to level up version of `dbt-sqlserver` in order to fix bug with `MAXRECURSION` error in dbt_run.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -8,9 +8,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
 
 ### Changed
 
+### Removed
+- Removed dbt-core==1.3.2 from requirements.txt
```
Comment (Rafalz13) [inline CHANGELOG.md:16]:
```suggestion
- Removed `dbt-core==1.3.2` from `requirements.txt`.
```

Comment (adrian-wojcik) [reply CHANGELOG.md]: Done

File: CHANGELOG.md (hunk 2 above)
Comment (adrian-wojcik) [reply CHANGELOG.md]: Done

**Review by Rafalz13 — APPROVED** [review]

**Pull request merged**